---
layout: page
title: Syllabus
subtitle: Transcriptomics
---

## 1. GENERAL INFORMATION

1.1. Course Name: Topics in Biological Sciences V (Section 1: Transcriptomics)  
1.2. Course Type: Specialization  
1.3. Professor: Dr. Tatiana Teixeira Torres, e-mail: tttorres(bio)ib.usp.br  
replace (bio) with @   
1.4. Program: Doctorate in Biological Sciences  
1.5. Specialization: --  
1.6. Course Code: B3P1Z1B6  
1.7. Credits: 2.0  
1.8. Weekly Hours: 16 hours  
1.9. Total Semester Hours: 32 hours  
1.10. Academic Semester: 2024-II  
1.11. Duration: 2 weeks  
1.12. Start Date: January 24, 2025  
1.13. End Date: January 31, 2025  
1.14. Location and Classroom: Computer Center 411 - Teaching Pavilion, Faculty of Biological Sciences, UNMSM  
1.15. Schedule:  
Monday, Tuesday, Thursday, and Friday:  
10:00 AM to 12:00 PM and 2:00 PM to 6:00 PM  
  
Enlace para el documento en español: ![Sílabo](/files/B3P1Z1B6silabo.pdf)  

---

## 2. COURSE FOUNDATIONS

### 2.1 Summary  
This course corresponds to the advanced period of study, is theoretical-practical in nature, and is delivered in person. Its purpose is to address and analyze topics of current interest in Biological Sciences.  
I. Introduction.  
II. Development.  
III. Analysis.  
IV. Applications.  

### 2.2 General Competence  
Ability to learn and continuously update knowledge.  

### 2.3 Specific Competences 
Applies methodologies to solve research problems in various areas of biological knowledge, with special emphasis on biodiversity and molecular biology.  

---
 
## 3. THEMATIC CONTENT   

### 3.1 Learning Unit I: Introduction    
- Introduction to transcriptomics and its relevance in molecular and evolutionary biology.     
- Overview of gene expression regulation. Types of transcriptomes: total RNA, messenger RNA, small RNAs.  
- Introduction to RNA-Seq: basic concepts and data structures.                           

### 3.2 Learning Unit II: Development  
- Types of transcriptomic and gene expression analyses: microarrays, qPCR, RNA-Seq, Iso-Seq, ATAC-Seq.   
- Data preprocessing: quality control and read filtering.                                   
- De novo assembly vs. reference-based genome alignment.            
- Gene expression quantification methods: RPKM, FPKM, TPM.                                 
- Statistical concepts: hypotheses, multiple testing, FDR.                                 
- Functional annotation of transcripts (Gene Ontology, KEGG) and functional enrichment analysis methods.  
- Integration of transcriptomics with genomics and proteomics.                             

### 3.3 Learning Unit III: Analysis  
- Downloading and organizing raw RNA-Seq data from public repositories (e.g., SRA).           
- Use of quality control tools (FastQC, Trimmomatic).                                     
- Transcriptome assembly (de novo with Trinity).                                  
- Transcript quantification with Salmon or Kallisto.                                        
- Initial visualization of expression data.                                                
- Differential expression analysis with DESeq2 in R and identification/interpretation of differentially expressed genes.  
- Transcript annotation with Blast2GO or similar tools.                                        

### 3.4 Learning Unit IV: Applications  
- Students present potential applications of transcriptomic and gene expression analysis in their projects.  
- Application of methods to compare gene expression profiles.                               
- Studies applying differential analyses to identify genes associated with specific conditions.  
- Functional enrichment in transcriptomic analyses.                                          
- Integration of transcriptomic data into a multi-omics study. 

---

## 4. METHODOLOGICAL STRATEGIES  
- Seminars  
- Problem-Based Learning (PBL)  

---

## 5. ASSESSMENT STRATEGIES  

### 5.1. Assessment Modalities:
- Exams, essays, projects, reports, etc.  

### 5.2. Evaluation Criteria:  
Weighting of each modality:  

| **Modality**                                | **Percentage** |  
|---------------------------------------------|----------------|  
| 1. Participation in article discussions     | 25%            |  
| 2. Participation in theoretical discussions | 25%            |  
| 3. Final report based on results of practical analyses conducted during the course. | 50% |  
| **Total**                                   | **100%**       |  

### 5.3. Calculation of Final Grade:
Final grade = Average of 1 (25%) + 2 (25%) + 3 (50%)  
Minimum passing grade: 14  

### 5.4. Requirements to Pass the Course:
- Attendance is mandatory. Absences must be justified with supporting documents to count the absent grade in the final weighting. The justification must be submitted at least 48 hours before or no later than 48 hours after the absence.  
- Daily assessments cannot be retaken due to absence (justified or not).  
- Unjustified absences exceeding 30% disqualify the student from continuing the course, resulting in a final grade of ZERO (0).  

---

### BIBLIOGRAPHY 
1. Griffiths, A. J. F., Wessler, S. R., Carroll, S. B., & Doebley, J. (2015). Regulation of gene expression in eukaryotes. In A. J. F. Griffiths, S. R. Wessler, S. B. Carroll, & J. Doebley (Eds.), *Introduction to Genetic Analysis* (11th ed., Chapter 12). New York: W.H. Freeman.  
2. Griffiths, A. J. F., Wessler, S. R., Carroll, S. B., & Doebley, J. (2015). Genomes and genomics. In A. J. F. Griffiths, S. R. Wessler, S. B. Carroll, & J. Doebley (Eds.), *Introduction to Genetic Analysis* (11th ed., Chapter 14). New York: W.H. Freeman.  
3. Lowe, R., Shirley, N., Bleackley, M., Dolan, S., & Shafee, T. (2017). Transcriptomics technologies. *PLOS Computational Biology*, 13(5), e1005457. https://doi.org/10.1371/journal.pcbi.1005457  
4. Stark, R., Grzelak, M., & Hadfield, J. (2019). RNA sequencing: The teenage years. *Nature Reviews Genetics*, 20(11), 631–656. https://doi.org/10.1038/s41576-019-0150-2  
5. Deshpande, D., Chhugani, K., Chang, Y., Karlsberg, A., Loefler, C., Zhang, J., Muszyńska, A., Munteanu, V., Yang, H., Rotman, J., Tao, L., Balliu, B., Tseng, E., Eskin, E., Zhao, F., Mohammadi, P., Pärnabaj, P., & Mangul, S. (2023). RNA-seq data science: From raw data to effective interpretation. *Frontiers in Genetics*, 14, 997383. https://doi.org/10.3389/fgene.2023.997383  

