---
layout: post
title: Day 03
#image: /img/hello_world.jpeg
---

**Paper for discussion in class**: 

- **28/01**: Mortazavi et al. (2008) Mapping and quantifying mammalian transcriptomes by RNA-Seq. Nat Methods, **5**, 621â€“628. https://doi.org/10.1038/nmeth.1226 [[pdf]](../pdf/Mortazavietal2008.pdf)
- **29/01**: Herbert et al. (2024) Ancient developmental genes underlie evolutionary novelties in walking fish. Curr Biol, **34**, 4339-4348. https://doi.org/10.1016/j.cub.2024.08.042 [[pdf]](../pdf/Herbertetal2024.pdf) 

---

### Assembly of RNA-Seq reads using Trinity PART I: Running Trinity
  
#### Lecture
[RNA-Seq assembly I](../pdf/Day03-A.pdf)

#### Running Trinity

1. Activate conda environment  
```
source activate
# source ~/miniconda3/bin/activate
conda activate trinity
```
  
2. Set species name  
```
# Basename
SPECIES="Chom"
```
  
3. Command to run Trinity 
```  
# Running
Trinity --seqType fq \
        --left ~/rnaseq/02-FilteredReads/${SPECIES}_R1_paired.fastq \
        --right ~/rnaseq/02-FilteredReads/${SPECIES}_R2_paired.fastq  \
        --CPU 6 \
        --max_memory 6G \
        --output ${SPECIES}.trinity.fasta \
        --full_cleanup 
```


#### Files for today's class (from the oven):  

- All files are in Google drive: [link to files](https://drive.google.com/drive/folders/1lp6qrIan160p0PZX7-Fu_0COtPF2t7Yt?usp=share_link) 

- Calb assembly: MD5 = ab1a66810dccf2f2cc4c6b4fd34adc89
- Cbez assembly: MD5 = da3014c0444b028b9eb4952763e767e2
- Chom assembly: MD5 = 129a2eeb0a40476487afc09d0923f511
- Cmac assembly: MD5 = d1bbf943c3053420abab0b8c2284c0c7
- Cmeg assembly: MD5 = f85c44d05a6ef83cc50c121e9e4437bb
- Lexi assembly: MD5 = cad424d7c60741cc99ccd2f394cbc09f
      

&nbsp;
&nbsp;


---

### Using public datasets  
  
#### Lecture
[Public datasets](../pdf/Day03-B.pdf)
  
#### Databases and utilities:  

- Sequence Read Archive (SRA): [https://www.ncbi.nlm.nih.gov/sra](https://www.ncbi.nlm.nih.gov/sra)  
- Gene Expression Omnibus (GEO): [https://www.ncbi.nlm.nih.gov/geo/](https://www.ncbi.nlm.nih.gov/geo/)  
- SRA-Toolkit: [https://hpc.nih.gov/apps/sratoolkit.html](https://hpc.nih.gov/apps/sratoolkit.html)
- grabseqs: [https://github.com/louiejtaylor/grabseqs](https://github.com/louiejtaylor/grabseqs)
- Genotype-Tissue Expression (GTEx) Project: [https://www.gtexportal.org/](https://www.gtexportal.org/)
- The Cancer Genome Atlas (TCGA) project: [https://www.cancer.gov/ccg/research/genome-sequencing/tcga](https://www.cancer.gov/ccg/research/genome-sequencing/tcga)
- Genbank: [https://www.ncbi.nlm.nih.gov/nucleotide/] (https://www.ncbi.nlm.nih.gov/nucleotide/)
- UniProt: [https://www.uniprot.org](https://www.uniprot.org)
- Pfam: [http://pfam.xfam.org](http://pfam.xfam.org)

  
---
  
### Assembly of RNA-Seq reads using Trinity PART II: Understanding Trinity
  
#### Lecture
[RNA-Seq assembly II](../pdf/Day03-C.pdf)
  
---
  
### Annotation of assembled transcripts 
  
#### Lecture
[Annotation](../pdf/Day03-D.pdf)


#### Annotating the assembled transcripts

1. Download databases (Linux)
```
wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
wget hhttps://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz
```

~ 24 minutes

2. Prepare databases
```
gunzip uniprot_sprot.fasta.gz
gunzip Pfam-A.hmm.gz
makeblastdb -in uniprot_sprot.fasta -dbtype prot
hmmpress Pfam-A.hmm
```

3. Search a protein database such as Swissprot (fast) or Uniref90 (slow but more comprehensive) using BLAST+
```
blastp -query transdecoder_dir/longest_orfs.pep  \
       -db uniprot_sprot.fasta  -max_target_seqs 1 \
       -outfmt 6 -evalue 1e-5 -num_threads 8 > blastp.out
```



&nbsp;
&nbsp;
---
 




    
    
