---
layout: post
title: Day 05
#image: /img/hello_world.jpeg
---

**Paper for discussion in class** (never gonna happen ðŸ˜†): 

- **30/01**: Mortazavi et al. (2008) Mapping and quantifying mammalian transcriptomes by RNA-Seq. Nat Methods, **5**, 621â€“628. https://doi.org/10.1038/nmeth.1226 [[pdf]](../pdf/Mortazavietal2008.pdf)
- **31/01**: Herbert et al. (2024) Ancient developmental genes underlie evolutionary novelties in walking fish. Curr Biol, **34**, 4339-4348. https://doi.org/10.1016/j.cub.2024.08.042 [[pdf]](../pdf/Herbertetal2024.pdf) 

&nbsp;  

---

### Mapping of reads to a reference database using Salmon

#### Lecture
[Mapping](../pdf/Day05-A.pdf)

#### Utilities:  

- Salmon: [https://combine-lab.github.io/salmon/](https://combine-lab.github.io/salmon/)
- MultiQC: [https://seqera.io/multiqc/](https://seqera.io/multiqc/)

&nbsp;   

#### Mapping

1\. Navigate to the `00-Databases` folder
```
cd ~/rnaseq/00-Databases
```

2\. Check if you have the reference databases of the mitochondrial genomes. If you haven't done it yet, download databases from the [course shared drive](https://drive.google.com/drive/folders/1lp6qrIan160p0PZX7-Fu_0COtPF2t7Yt?usp=share_link)  
```
ls -la
```

3\. Activate 'quant' environment
```
conda activate quant
```

4\. Call salmon to read help
```
salmon index -h
salmon quant -h
salmon quant --help-reads
```

5\. Create reference index for mtDNA CDSs
```
#Basename
SPECIES="Lexi"

#Building index
salmon index -t ~/rnaseq/00-Databases/${SPECIES}_mtDNAgenes.fasta \
             -i ~/rnaseq/00-Databases/${SPECIES}_mtDNAgenes_index 
```

6\. Create and navigate to the folder `05-Mapping`
```
mkdir ~/rnaseq/05-Mapping
cd ~/rnaseq/05-Mapping
```

7\. Run salmon for quantification 
```
#Basename
SPECIES="Calb"
SAMPLE="F_1"

salmon quant \
       -i ~/rnaseq/00-Databases/${SPECIES}_mtDNAgenes_index \
       -l A \
       -r ~/rnaseq/01-RawReads/${SPECIES}${SAMPLE}_R1.fastq.gz \
       -o ~/rnaseq/05-Mapping/${SPECIES}${SAMPLE}_mtDNA-salmon
```


&nbsp;  


---

#### Script in bash for many species

```
#!/bin/bash

# Defining species
SPECIES="Lexi"

# List of samples
SAMPLES=("F_1" "F_2" "F_3" "L_1" "L_2" "L_3")

# Paths
INDEX_DIR="/home/informatica/rnaseq/00-Databases/${SPECIES}_mtDNAgenes_index"
READS_DIR="/home/informatica/rnaseq/01-RawSeqs/${SPECIES}"
OUTPUT_DIR="/home/informatica/rnaseq/05-Mapping/"

# Loop
for SAMPLE in "${SAMPLES[@]}"; do
  echo "Processing sample: ${SAMPLE}"
  
  # Running salmon
  salmon quant \
    -i "${INDEX_DIR}" \
    -l A \
    -r "${READS_DIR}/${SPECIES}${SAMPLE}_R1.fastq" \
    -o "${OUTPUT_DIR}/${SPECIES}${SAMPLE}_mtDNA-salmon"
done

echo "Done!"
```

---

### Differential Gene Expression Analysis in DESeq2

#### Lecture
[DGE](../pdf/Day05-B.pdf)

#### R script:  
[Rscript](../files/DGEmtDNA.R)
