---
layout: post
title: Day 04
#image: /img/hello_world.jpeg
---

**Paper for discussion in class**: 

- **29/01**: Mortazavi et al. (2008) Mapping and quantifying mammalian transcriptomes by RNA-Seq. Nat Methods, **5**, 621â€“628. https://doi.org/10.1038/nmeth.1226 [[pdf]](../pdf/Mortazavietal2008.pdf)
- **30/01**: Herbert et al. (2024) Ancient developmental genes underlie evolutionary novelties in walking fish. Curr Biol, **34**, 4339-4348. https://doi.org/10.1016/j.cub.2024.08.042 [[pdf]](../pdf/Herbertetal2024.pdf) 

---

### Annotation of assembled transcripts 

#### Downloading databases for today's activities

1\. Create a new folder and navigate to it  
   
```
mkdir ~/rnaseq/00-Databases/
```

2\. Download databases
   
```
wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
wget https://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz
```
> 10-20 minutes

&nbsp;  
#### Lecture
[Annotation](../pdf/Day04-A.pdf)

#### Databases and utilities:  

- Genbank: [https://www.ncbi.nlm.nih.gov/nucleotide/](https://www.ncbi.nlm.nih.gov/nucleotide/)
- UniProt: [https://www.uniprot.org](https://www.uniprot.org)
- ORFfinder: [https://www.ncbi.nlm.nih.gov/orffinder/](https://www.ncbi.nlm.nih.gov/orffinder/)
- Pfam: [http://pfam.xfam.org](http://pfam.xfam.org)
- DeepTMHMM: [https://services.healthtech.dtu.dk/services/DeepTMHMM-1.0/](https://services.healthtech.dtu.dk/services/DeepTMHMM-1.0/)
- SignalP: [https://services.healthtech.dtu.dk/services/SignalP-6.0/](https://services.healthtech.dtu.dk/services/SignalP-6.0/)
- eggNOG-mapper: [http://eggnog-mapper.embl.de](http://eggnog-mapper.embl.de)
- TransDecoder: [https://github.com/TransDecoder/TransDecoder/wiki](https://github.com/TransDecoder/TransDecoder/wiki)
- Docker: [https://docs.docker.com/desktop/setup/install/linux/ubuntu/](https://docs.docker.com/desktop/setup/install/linux/ubuntu/)
- Trinotate: [https://github.com/Trinotate/Trinotate/wiki/TrinotateViaDocker](https://github.com/Trinotate/Trinotate/wiki/TrinotateViaDocker)


#### Sequence for Pfam search

```
>FBpp0072038
MKEFQRMLMLQYSKHGECILKEIGAAFRGEHPADLTIVCENKVKLHAHKLVLAAASPLIR
NLLEDTHLSDCSTTVYFPDVNATYFKFLLDFLYSGQTCITSRDVNYLHDLLLLLQIKSDS
WKTTDSAYLSSKCGGLRDRADRRKQQYTSPQNLEPDQTLKYEVDSVDESRNAADFSSAFN
SNDNCESAAECERSGGHNNKEEDEDDCTHKDNKSDKDTDEIVNLSNAPPSGTSGSNSNIS
TSSNHQQQQHHHHHHHNHNNNNNNNNNNSSSSTINPVNLSLDLRTKSENSASRTLGSGSD
HSGIDLAVTASESTKRKGLFFDSHKDVMKPLSDGSDINSSPENYVVTPHRKRRPGFHNTQ
SDNQPFTSYPHSLLEELRLAKSTTSPISGFGSEKNMLAHLEDGALNGDTLTPDRKHLLEA
QRNRAQSPEIPMHLGPQFVYQWQSNQNAAMSAMPNLQSRLSSLSHISLNLDHPEGRSGSA
SGSGANLAGSNTHASSVREYRCEYCGKQFGMSWNLKTHLRVHTGEKPFACRLCVAMFKQK
AHLLKHLCSVHRNVITTTNGADTENRYSCCFCSMCFESVQELVRHLSGHHNNLLLTKNLR
E
```

#### Sequence for prediction of transmembrane helices in proteins

```
>FBpp0290693
MSSNEAYHEPGAGGDGPGGSSGASGGGSQRSNQLHHQQILNETTYLKPAAKQAYFSDEKV
LIPDDDSTNVGFSFRKLWAFTGPGFLMSIAYLDPGNIESDMQSGAAAKYKILWVLLWATV
LGLLMQRLAARLGVVTGLHLAEMCYRQYKRLPRWILWIMIEIAIIGSDMQEVIGTAIAIY
LLSNKVVPLWGGVLITIVDTFTFLFLDKYGLRKLEFLFGTLITIMAVSFGYEYIVSAPNQ
GEVLEGMFVPWCSNCNSNVLLQAVGVVGAVIMPHNLYLHSALVKSRDIDRRQTKKVSEAN
FYFFIEASVALFVSFIINLFVVAVFAHGMYGKTNNDVVEVCKDKSMYEDAKMSFVDNVNG
TAIIDADLYKGGLFLGCTFGAVAMYIWGVGILAAGQSSTMTGTYAGQFSMEGFLNLQWPR
WCRVLVTRCIAIIPTFCLAMFSKMEDLTSMNDILNAVMSLQLPFAAIPTIAFTSCAAIMG
EFVNGLGNKIVSILLTIVVIGVNLYFVVVQVENMEIKGGLLALVCIFAILYILFNLYLVI
HMAACMGNQRLMNSRWVQRFVLPSQNSFSIKNANSTYARIATSSDQEPEGLAGEDA
```

#### Sequence for prediction of Signal Peptide

```
>FBpp0292967
MVLKVPTSKVLLVLATLFAVAAMISSWMPQVAASPLAPTEYEQRRMMCSTGLSDVIQKIC
VSGTVALGDVFPNSFGKRRKRDLQNVTDLCCKSGGCTYRELLQYCKG
```

---

#### Annotating the assembled transcripts

1\. Activate conda environment
```
source ~/miniconda3/bin/activate
conda activate annotation
```

2\. Setting basename
```
# Basename
SPECIES="Chom"
```

3\. Fixing our Trinity names
```
cd ~/rnaseq/03-Assembly/
mv ${SPECIES}.trinity.fasta.Trinity.fasta ${SPECIES}.Trinity.fasta
mv ${SPECIES}.trinity.fasta.Trinity.fasta.gene_trans_map ${SPECIES}.Trinity.fasta.gene_trans_map
```   

4\. Running Transdecoder
```
TransDecoder.LongOrfs -t ~/rnaseq/03-Assembly/${SPECIES}.Trinity.fasta
```

5\. Prepare databases for BLAST and HMMER
```
cd ~/rnaseq/00-Databases/
gunzip uniprot_sprot.fasta.gz
gunzip Pfam-A.hmm.gz
makeblastdb -in uniprot_sprot.fasta -dbtype prot
hmmpress Pfam-A.hmm
```

6\. Search a protein database such as Swissprot (fast) or Uniref90 (slow but more comprehensive) using BLAST+
```
cd ~/rnaseq/04-Annotation/
blastp -query ${SPECIES}.Trinity.fasta.transdecoder_dir/longest_orfs.pep  \
       -db ~/rnaseq/00-Databases/uniprot_sprot.fasta  -max_target_seqs 1 \
       -outfmt 6 -evalue 1e-5 -num_threads 8 > blastp.out
```

7\. Search for functional domains using HMMER
```
hmmscan --cpu 8 --domtblout ${SPECIES}.hmmscanPFAM.out \
        ~/rnaseq/00-Databases/Pfam-A.hmm \
        ${SPECIES}.Trinity.fasta.transdecoder_dir/longest_orfs.pep
```

8a\. Prepare the Trinotate SQLite database in conda  
```
Trinotate --create \
          --db ${SPECIES}_Trinotate.sqlite \
          --trinotate_data_dir ~/rnaseq/04-Annotation/\
          ${SPECIES}/Trinotate
```

8b\. Prepare the Trinotate SQLite database in docker  
```
$TRINOTATE_HOME/Trinotate --create \
          --db ${SPECIES}_Trinotate.sqlite \
          --trinotate_data_dir /data/rnaseq/04-Annotation/\
          ${SPECIES}/Trinotate
```

9a\. Integrating results (conda)  
```
Trinotate --db ${SPECIES}_Trinotate.sqlite --init \
--gene_trans_map /data/rnaseq/03-Assembly/${SPECIES}/${SPECIES}.Trinity.fasta.gene_trans_map \
--transcript_fasta /data/rnaseq/03-Assembly/${SPECIES}/${SPECIES}.Trinity.fasta \
--transdecoder_pep /data/rnaseq/04-Annotation/${SPECIES}/transdecoder/longest_orfs.pep 
```

9b\. Integrating results (docker)  
```
$TRINOTATE_HOME/Trinotate --db ${SPECIES}_Trinotate.sqlite --init \
--gene_trans_map /data/rnaseq/03-Assembly/${SPECIES}/${SPECIES}.Trinity.fasta.gene_trans_map \
--transcript_fasta /data/rnaseq/03-Assembly/${SPECIES}/${SPECIES}.Trinity.fasta \
--transdecoder_pep /data/rnaseq/04-Annotation/${SPECIES}/transdecoder/longest_orfs.pep 
```

10\. Loading BLAST results against SwissProt database   
```
$TRINOTATE_HOME/Trinotate \
             --db ${SPECIES}_Trinotate.sqlite \
             --LOAD_swissprot_blastp blastp.out
```

11\. Loading HMMScan results   
```
$TRINOTATE_HOME/Trinotate \
             --db ${SPECIES}_Trinotate.sqlite \
             --LOAD_pfam hmmscanPFAM.out
```

12\. Check if the database was populated  
```
$TRINOTATE_HOME/Trinotate \
             --db ${SPECIES}_Trinotate.sqlite \ 
             --report >${SPECIES}_TrinotateReport.xls
            
```

#### Installing Trinotate with Docker
1\. Download [Docker](https://docs.docker.com/desktop/setup/install/linux/ubuntu/).
2\. Follow the instructions for installation.
3\. Open your Applications menu and search for Docker Desktop.
4\. Select Docker Desktop to start Docker. You'll need to create a login.
5\. Select Accept to continue. Docker Desktop starts after you accept the terms.
6\. On Terminal:  
```
docker --version
```  
7\. Download [Trinotate Docker image](https://github.com/Trinotate/Trinotate/wiki/TrinotateViaDocker):  
```
docker pull trinityrnaseq/trinotate
```    
8\. Run Trinotate interactively
```
docker run --rm -it \
                 -v `pwd`:/data \
                 -v /tmp:/tmp \
                 -e TRINOTATE_HOME=/usr/local/src/Trinotate \
                 trinityrnaseq/trinotate bash
```  
9\. Test Trinotate  
```
$TRINOTATE_HOME/Trinotate 
```  



#### Files for today's class (from the oven):  

- All files are in Google drive: [link to files](https://drive.google.com/drive/folders/1lp6qrIan160p0PZX7-Fu_0COtPF2t7Yt?usp=share_link) 

- Calb assembly: MD5 = ab1a66810dccf2f2cc4c6b4fd34adc89 
- Cbez assembly: MD5 = da3014c0444b028b9eb4952763e767e2  
- Chom assembly: MD5 = 129a2eeb0a40476487afc09d0923f511  
- Cmac assembly: MD5 = d1bbf943c3053420abab0b8c2284c0c7  
- Cmeg assembly: MD5 = f85c44d05a6ef83cc50c121e9e4437bb  
- Lexi assembly: MD5 = cad424d7c60741cc99ccd2f394cbc09f  

**Databases**
- uniprot_sprot.fasta.gz: MD5 = 76e3d7039e39b9e67c0038b6460a5443  
- Pfam-A.hmm.gz: MD5 = 768b7d9e42fb08af15cc578f15a89cf9  

&nbsp;
&nbsp;


---

### Mapping reads against a reference using Salmon

#### Lecture
[Mapping](../pdf/Day04-B.pdf)

#### Downloading raw reads for tomorrow's activities

- All files are in Google drive: [link to files](https://drive.google.com/drive/folders/1LtVsOMqkkU3YqFjR24Q78YnzXIw1qdX8?usp=drive_link) 

MD5 (CalbF_1_R1.fastq.gz) = c7a10da327051bcc2b2e2f3c3cc6962f  
MD5 (CalbF_2_R1.fastq.gz) = b6fe1ed37425e564a2eb65d6e2dfcc26  
MD5 (CalbF_4_R1.fastq.gz) = 039b74c5465bd3783fe75f6105a95cd1  
MD5 (CalbL_1_R1.fastq.gz) = 293f2e0c5e38d8bd06b4bdeb172c3bf4  
MD5 (CalbL_2_R1.fastq.gz) = 306817bf00ec85e47c364e213fbc5b1d  
MD5 (CalbL_4_R1.fastq.gz) = e329f6e1ceaac55809f479af2d244be4  
  
MD5 (CbezF_1_R1.fastq.gz) = d9cfd74fec47a8ac32e580d0f2d95e0f  
MD5 (CbezL_1_R1.fastq.gz) = e0a103a41257ce83294ac93c2ac9844c  
  
MD5 (ChomF_1_R1.fastq.gz) = 8181a8d5a5043164ccaba5b3d76650c8  
MD5 (ChomF_2_R1.fastq.gz) = 99a613201926de1a45f7c4ffd6bab994  
MD5 (ChomF_3_R1.fastq.gz) = 27b4ec04a0a4478eeeb2266858b5f2d8  
MD5 (ChomL_1_R1.fastq.gz) = 9a3abb89c6c77bf55c4f15d3e8f9057f  
MD5 (ChomL_2_R1.fastq.gz) = 01274d802cffffa608ab47496d53cd26  
MD5 (ChomL_3_R1.fastq.gz) = 8b3bbf34eccc1f481a2286c0e8e19720  
  
MD5 (CmacF_1_R1.fastq.gz) = 5130110bb7465ba7ad371a3735610e12  
MD5 (CmacF_2_R1.fastq.gz) = 7bf77683aa19dcbc192dee2dc570ed84  
MD5 (CmacF_3_R1.fastq.gz) = 7ec36657ed29b98eed92ac00d242e272  
MD5 (CmacL_1_R1.fastq.gz) = 6a679a6ddf48e328b49ddaa21fe02c69  
MD5 (CmacL_2_R1.fastq.gz) = 8e30017c41981a5ca74c640bf85f053e  
MD5 (CmacL_3_R1.fastq.gz) = 52300e3883bf0c0d8e84e628e257b603  
  
MD5 (CmegF_1_R1.fastq.gz) = f42d69a06fdc1a42ba6111b80145f2ed  
MD5 (CmegF_2_R1.fastq.gz) = d0b183d892c0778e3ed22a36727a9d19  
MD5 (CmegF_4_R1.fastq.gz) = f056cb4dd147151cdb428afe51c1a018  
MD5 (CmegL_1_R1.fastq.gz) = 4e19a9ada229ef8eee6f77548b9836c6  
MD5 (CmegL_2_R1.fastq.gz) = 563870dce7e2a1329dc07d1aaa54bfba  
MD5 (CmegL_4_R1.fastq.gz) = dbc635f44bde52ea4eec0a1fdb6ccd62  
  
MD5 (LexiF_1_R1.fastq.gz) = 6f628f653152c69841bf78703cec5a6b  
MD5 (LexiF_2_R1.fastq.gz) = f4f41f3b8409831393bd0bea59749a2f  
MD5 (LexiF_3_R1.fastq.gz) = 897da1de9949c0ce0b8e6081f06bbb13  
MD5 (LexiL_1_R1.fastq.gz) = 3306fbcb5c5b8df94a15b612022f3f73  
MD5 (LexiL_2_R1.fastq.gz) = f826424a63f2c44edd6f83438c213e9e  
MD5 (LexiL_3_R1.fastq.gz) = 92152c9bb385b40372043398b29b474d  
    

&nbsp;
&nbsp;

---
 
