---
layout: post
title: Day 04
#image: /img/hello_world.jpeg
---

**Paper for discussion in class**: 

- **29/01**: Mortazavi et al. (2008) Mapping and quantifying mammalian transcriptomes by RNA-Seq. Nat Methods, **5**, 621â€“628. https://doi.org/10.1038/nmeth.1226 [[pdf]](../pdf/Mortazavietal2008.pdf)
- **30/01**: Herbert et al. (2024) Ancient developmental genes underlie evolutionary novelties in walking fish. Curr Biol, **34**, 4339-4348. https://doi.org/10.1016/j.cub.2024.08.042 [[pdf]](../pdf/Herbertetal2024.pdf) 

---

### Annotation of assembled transcripts 

#### Downloading databases for today's activities

1. Create a new folder and navigate to it
```
mkdir ~/rnaseq/00-Databases/
cd ~/rnaseq/00-Databases/
```

2. Download databases
```
wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
wget hhttps://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz
```
> 20 minutes
  
#### Lecture
[Annotation](../pdf/Day04-A.pdf)

#### Databases and utilities:  

- Genbank: [https://www.ncbi.nlm.nih.gov/nucleotide/](https://www.ncbi.nlm.nih.gov/nucleotide/)
- UniProt: [https://www.uniprot.org](https://www.uniprot.org)
- ORFfinder: [https://www.ncbi.nlm.nih.gov/orffinder/](https://www.ncbi.nlm.nih.gov/orffinder/)
- Pfam: [http://pfam.xfam.org](http://pfam.xfam.org)
- DeepTMHMM: [https://services.healthtech.dtu.dk/services/DeepTMHMM-1.0/](https://services.healthtech.dtu.dk/services/DeepTMHMM-1.0/)
- SignalP: [https://services.healthtech.dtu.dk/services/SignalP-6.0/](https://services.healthtech.dtu.dk/services/SignalP-6.0/)
- eggNOG-mapper: [http://eggnog-mapper.embl.de](http://eggnog-mapper.embl.de)
- TransDecoder: [https://github.com/TransDecoder/TransDecoder/wiki](https://github.com/TransDecoder/TransDecoder/wiki)


#### Sequence for Pfam search
```
>FBpp0072038
MKEFQRMLMLQYSKHGECILKEIGAAFRGEHPADLTIVCENKVKLHAHKLVLAAASPLIR
NLLEDTHLSDCSTTVYFPDVNATYFKFLLDFLYSGQTCITSRDVNYLHDLLLLLQIKSDS
WKTTDSAYLSSKCGGLRDRADRRKQQYTSPQNLEPDQTLKYEVDSVDESRNAADFSSAFN
SNDNCESAAECERSGGHNNKEEDEDDCTHKDNKSDKDTDEIVNLSNAPPSGTSGSNSNIS
TSSNHQQQQHHHHHHHNHNNNNNNNNNNSSSSTINPVNLSLDLRTKSENSASRTLGSGSD
HSGIDLAVTASESTKRKGLFFDSHKDVMKPLSDGSDINSSPENYVVTPHRKRRPGFHNTQ
SDNQPFTSYPHSLLEELRLAKSTTSPISGFGSEKNMLAHLEDGALNGDTLTPDRKHLLEA
QRNRAQSPEIPMHLGPQFVYQWQSNQNAAMSAMPNLQSRLSSLSHISLNLDHPEGRSGSA
SGSGANLAGSNTHASSVREYRCEYCGKQFGMSWNLKTHLRVHTGEKPFACRLCVAMFKQK
AHLLKHLCSVHRNVITTTNGADTENRYSCCFCSMCFESVQELVRHLSGHHNNLLLTKNLR
E
```

#### Sequence for prediction of transmembrane helices in proteins
```
>FBpp0290693
MSSNEAYHEPGAGGDGPGGSSGASGGGSQRSNQLHHQQILNETTYLKPAAKQAYFSDEKV
LIPDDDSTNVGFSFRKLWAFTGPGFLMSIAYLDPGNIESDMQSGAAAKYKILWVLLWATV
LGLLMQRLAARLGVVTGLHLAEMCYRQYKRLPRWILWIMIEIAIIGSDMQEVIGTAIAIY
LLSNKVVPLWGGVLITIVDTFTFLFLDKYGLRKLEFLFGTLITIMAVSFGYEYIVSAPNQ
GEVLEGMFVPWCSNCNSNVLLQAVGVVGAVIMPHNLYLHSALVKSRDIDRRQTKKVSEAN
FYFFIEASVALFVSFIINLFVVAVFAHGMYGKTNNDVVEVCKDKSMYEDAKMSFVDNVNG
TAIIDADLYKGGLFLGCTFGAVAMYIWGVGILAAGQSSTMTGTYAGQFSMEGFLNLQWPR
WCRVLVTRCIAIIPTFCLAMFSKMEDLTSMNDILNAVMSLQLPFAAIPTIAFTSCAAIMG
EFVNGLGNKIVSILLTIVVIGVNLYFVVVQVENMEIKGGLLALVCIFAILYILFNLYLVI
HMAACMGNQRLMNSRWVQRFVLPSQNSFSIKNANSTYARIATSSDQEPEGLAGEDA
```

#### Sequence for prediction of Signal Peptide
```
>FBpp0292967
MVLKVPTSKVLLVLATLFAVAAMISSWMPQVAASPLAPTEYEQRRMMCSTGLSDVIQKIC
VSGTVALGDVFPNSFGKRRKRDLQNVTDLCCKSGGCTYRELLQYCKG
```

---

#### Annotating the assembled transcripts

1. Activate conda environment
```
source ~/miniconda3/bin/activate
conda activate annotation
```

2. Setting basename
```
# Basename
SPECIES="Chom"
```
3. Fixing our Trinity names
```
cd ~/rnaseq/03-Assembly/
mv ${SPECIES}.trinity.fasta.Trinity.fasta ${SPECIES}.Trinity.fasta
mv ${SPECIES}.trinity.fasta.Trinity.fasta.gene_trans_map ${SPECIES}.Trinity.fasta.gene_trans_map
```   

4. Running Transdecoder
```
TransDecoder.LongOrfs -t ~/rnaseq/03-Assembly/${SPECIES}.Trinity.fasta
```


2. Prepare databases
```
gunzip uniprot_sprot.fasta.gz
gunzip Pfam-A.hmm.gz
makeblastdb -in uniprot_sprot.fasta -dbtype prot
hmmpress Pfam-A.hmm
```

3. Search a protein database such as Swissprot (fast) or Uniref90 (slow but more comprehensive) using BLAST+
```
blastp -query transdecoder_dir/longest_orfs.pep  \
       -db uniprot_sprot.fasta  -max_target_seqs 1 \
       -outfmt 6 -evalue 1e-5 -num_threads 8 > blastp.out
```




#### Files for today's class (from the oven):  

- All files are in Google drive: [link to files](https://drive.google.com/drive/folders/1lp6qrIan160p0PZX7-Fu_0COtPF2t7Yt?usp=share_link) 

- Calb assembly: MD5 = ab1a66810dccf2f2cc4c6b4fd34adc89
- Cbez assembly: MD5 = da3014c0444b028b9eb4952763e767e2
- Chom assembly: MD5 = 129a2eeb0a40476487afc09d0923f511
- Cmac assembly: MD5 = d1bbf943c3053420abab0b8c2284c0c7
- Cmeg assembly: MD5 = f85c44d05a6ef83cc50c121e9e4437bb
- Lexi assembly: MD5 = cad424d7c60741cc99ccd2f394cbc09f
      

&nbsp;
&nbsp;


---

### Using public datasets  
  
#### Lecture
[Public datasets](../pdf/Day03-B.pdf)
  

  
---
  

  
---
  

&nbsp;
&nbsp;
---
 
