---
layout: post
title: Day 03
#image: /img/hello_world.jpeg
---

**Paper for discussion in class**: 

- **28/01**: Herbert et al. (2024) Ancient developmental genes underlie evolutionary novelties in walking fish. Curr Biol, **34**, 4339-4348. https://doi.org/10.1016/j.cub.2024.08.042 [[pdf]](../pdf/Herbertetal2024.pdf)

---

### Using public datasets  
  
#### Lecture
[Public datasets](../pdf/Day03-A.pdf)
  
#### Databases:  

- Sequence Read Archive (SRA): [https://www.ncbi.nlm.nih.gov/sra](https://www.ncbi.nlm.nih.gov/sra)  
- Gene Expression Omnibus (GEO): [https://www.ncbi.nlm.nih.gov/geo/](https://www.ncbi.nlm.nih.gov/geo/)  
- SRA-Toolkit: [https://hpc.nih.gov/apps/sratoolkit.html](https://hpc.nih.gov/apps/sratoolkit.html)

  
---
  
### Assembly of RNA-Seq reads using Trinity
  
#### Lecture
[RNA-Seq assembly](../pdf/Day03-B.pdf)
  
---
  
### Annotation of assembled transcripts 
  
#### Lecture
[Annotation](../pdf/Day03-C.pdf)

---

#### Files for today's class:  

- All files are in Google drive: [link to files](https://drive.google.com/drive/folders/1lp6qrIan160p0PZX7-Fu_0COtPF2t7Yt?usp=share_link) 

- Calb Female R1: MD5 = d0ae3be816d3497324173dbc49143186
- Calb Female R2: MD5 = 4ca2fdd31ce8f01c9c1da9606ac284f0

- Calb Larvae R1: MD5 = 3662b33a198ef8aec12cdb9864131086
- Calb Larvae R2: MD5 = 2db2e4f1c805c61e240b14ad4f8c5413
  
- Cbez Female R1: MD5 = 23352efdaba107b63bc84ad43f0643c8
- Cbez Female R2: MD5 = 1974b22111651d26804f945d003d4a7c
  
- Cbez Larvae R1: MD5 = 46c4ac46d12ea1288f1a90a0c22f7142
- Cbez Larvae R2: MD5 = 8c045942c6296b0e3ac61df5c99d483c
  
- Chom Female R1: MD5 = fcdae866b748d6cd7d9052f5d22c2cb1
- Chom Female R2: MD5 = 8fc1bcbbac214b70f332b248684a7516
  
- Chom Larvae R1: MD5 = d3ceea3a8297ec8c1ff937bd0caebaa5
- Chom Larvae R2: MD5 = a4bbac4f7bdc158fb1b524aca69197e2
  
- Cmac Female R1: MD5 = cab11dcbefaa8f382fbb78ec58d49af0
- Cmac Female R2: MD5 = 4943a71e10951450ce4ad814d29defa4
  
- Cmac Larvae R1: MD5 = e13cf8812372d0b1b06b2fdc8de526a1
- Cmac Larvae R2: MD5 = 435d66e0fc41dc720b08f3c49128e01a
  
- Cmeg Female R1: MD5 = e8d4b752411c30c55108f288b9cb25ca
- Cmeg Female R2: MD5 = efefa98031228ea34f0a15d3ded6f5bf
  
- Cmeg Larvae R1: MD5 = 3ca78d903c705a8982ac6f3eda2d41c2
- Cmeg Larvae R2: MD5 = 9c17f124ca73c85cba95b0722bb5fe38  
  
- Lexi R1: MD5 = a01fc0297f330d2846daef7b47bedad3
  
- Lexi R2: MD5 = ecc597a302a3258e27f21d5e2a21286c
   

---    

#### Running Trimmomatic

1. File with adapter sequences: [TruSeq3-PE.fa](../files/TruSeq3-PE.fa)  
  
2. Species name  
```
# Basename
SPECIES="Chom"
```
  
3. Command to run Trimmomatic  
```  
# Running
trimmomatic PE \
            -threads 8 \
            -phred33 \
            -summary ${SPECIES}-report.txt \
            ~/rnaseq/01-RawReads/${SPECIES}_R1.fastq.gz \
            ~/rnaseq/01-RawReads/${SPECIES}_R2.fastq.gz \
            ${SPECIES}_R1_paired.fastq ${SPECIES}_R1_unpaired.fastq \
            ${SPECIES}_R2_paired.fastq ${SPECIES}_R2_unpaired.fastq \
            ILLUMINACLIP:../TruSeq3-PE.fa:2:30:10:2:True \
            SLIDINGWINDOW:4:20 \
            LEADING:3 \
            TRAILING:3 \
            MINLEN:50
```
  
4. Remove *unpaired.fastq files  
```
rm *_unpaired.fastq
```

&nbsp;
&nbsp;

---


    
    
